app/src/main/kotlin/org/eurofurence/connavigator/ui/LoginActivity.kt

```diff
@@ -100,61 +107,88 @@ class LoginUi : AnkoComponent<LoginActivity> {
     lateinit var errorText: TextView
     lateinit var submit: Button
     lateinit var logout: Button
+    lateinit var layoutMain: LinearLayout
+    lateinit var layoutBusy: LinearLayout
 
     override fun createView(ui: AnkoContext<LoginActivity>) = with(ui) {
         linearLayout {
             lparams(matchParent, matchParent)
-            // If not logged in
-            verticalLayout {
-                visibility = if (AuthPreferences.isLoggedIn()) View.GONE else View.VISIBLE
 
-                toolbar {
-                    title = "Login"
-                    lparams(matchParent, wrapContent)
-                    backgroundResource = R.color.primary
-                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-                        setTitleTextColor(ctx.getColor(R.color.textWhite))
-                    }
+            layoutBusy = verticalLayout {
+                visibility = View.GONE
+                gravity = Gravity.CENTER
+                orientation = LinearLayout.VERTICAL
+
+                progressBar {
+                    lparams(wrapContent, wrapContent)
                 }
 
-                lparams(matchParent, matchParent)
+                textView("Authenticating...") {
+                    padding = 20
+                    gravity = Gravity.CENTER
 
-                regNumber = editText {
-                    hint = "Registration number"
-                    inputType = InputType.TYPE_CLASS_NUMBER
                     lparams(matchParent, wrapContent)
                 }
 
-                username = editText {
-                    hint = "Username"
-                    lparams(matchParent, wrapContent)
-                }
+                lparams(matchParent, matchParent)
+            }
 
-                password = editText {
-                    hint = "Password"
-                    inputType = InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD
-                    lparams(matchParent, wrapContent)
-                }
+            layoutMain = verticalLayout {
 
-                errorText = textView("Your login was unsuccessful, are you sure you entered the correct data?") {
-                    visibility = View.GONE
-                }
+                // If not logged in
+                verticalLayout {
+                    visibility = if (AuthPreferences.isLoggedIn()) View.GONE else View.VISIBLE
 
-                submit = button {
-                    text = "Submit"
-                    lparams(matchParent, wrapContent)
+                    toolbar {
+                        title = "Login"
+                        lparams(matchParent, wrapContent)
+                        backgroundResource = R.color.primary
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
use setTextAppearance(ctx, android.R.style.TextAppearance_Inverse). This code will only render the styles if it is compiled for version 6.0+. If pushed to beta or production it won't work